zingo-sync/src/scan.rs

```diff
@@ -101,27 +98,65 @@ impl InitialScanData<'_> {
                             frontiers.final_orchard_tree().tree_size() as u32,
                         )
                     }
-                    Ordering::Equal => (0, 0),
-                    Ordering::Less => panic!("pre-sapling not supported!"),
+                    cmp::Ordering::Equal => (0, 0),
+                    cmp::Ordering::Less => panic!("pre-sapling not supported!"),
                 }
             };
 
         Ok(InitialScanData {
-            previous_block,
+            previous_block: previous_wallet_block,
             sapling_initial_tree_size,
             orchard_initial_tree_size,
         })
     }
 }
 
+#[allow(dead_code)]
+struct ScanData {
+    pub(crate) nullifiers: NullifierMap,
+    pub(crate) wallet_blocks: BTreeMap<BlockHeight, WalletBlock>,
+    pub(crate) relevent_txids: HashSet<TxId>,
+    pub(crate) decrypted_note_data: DecryptedNoteData,
+    pub(crate) shard_tree_data: ShardTreeData,
+}
+
+pub(crate) struct ScanResults {
```

Comment:
please include comments explanating the methodology and significance of these results